### PR TITLE
Add support for deployment statuses

### PR DESCRIPTION
**Pull request by colbygallup**
Fixes issue #1586

### CODE REVIEW

**Comment by colbygallup** [comment]
@s-t-e-v-e-n-k I added the requested test. Need any other changes?

File: tests/DeploymentStatus.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+
+class DeploymentStatus(Framework.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.deployment = (
+            self.g.get_user().get_repo("PyGithub").get_deployment(242997115)
+        )
+        self.status = next(
+            s for s in self.deployment.get_statuses() if s.id == 344110026
```
Comment (s-t-e-v-e-n-k) [inline tests/DeploymentStatus.py:37]: There's a handy get_status() method, and you already have the id, there's no need to loop through them.

You should also add a test for get_statuses() though -- just asserting on the returned ids would be enough, such as the Workflow.testGetRunsWithObjects unit test.

Comment (colbygallup) [reply tests/DeploymentStatus.py:37]: I added a `Deployment.get_status()` before, but removed it once I realized that it was part of a feature preview:
```python
gh.get_user().get_repo("PyGithub").get_deployment(242997115).get_status("344427441") 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\colby\PyGithub\github\Deployment.py", line 162, in get_status
    "GET", self.url + "/statuses/" + str(id_)
  File "C:\Users\colby\PyGithub\github\Requester.py", line 319, in requestJsonAndCheck
    verb, url, parameters, headers, input, self.__customConnection(url)
  File "C:\Users\colby\PyGithub\github\Requester.py", line 342, in __check
    raise self.__createException(status, responseHeaders, output)
github.GithubException.GithubException: 415 {"message": "If you would like to help us test the Deployments API enhancement during its preview period, you must specify a custom media type in the 'Accept' header. Please 
see the docs for full details.", "documentation_url": "https://developer.github.com/v3/repos/deployments/#get-a-single-deployment-status"}
```

Comment (s-t-e-v-e-n-k) [reply tests/DeploymentStatus.py:37]: We already support a lot of API previews, so feel free to add it -- Repository.create_label() can give you some ideas.

Comment (colbygallup) [reply tests/DeploymentStatus.py:37]: Sounds good. I have the new test done already; I'll add the `get_status()` code back in and it'll be good for another review.

**Pull request merged**